core/src/main/java/org/apache/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -170,7 +170,15 @@ public DynamicClusterImpl() {
     @Override
     public void init() {
         super.init();
-        sensors().set(NEXT_CLUSTER_MEMBER_ID, new NextClusterMemberIdSupplier());
+        initialiseMemberId();
+    }
+
+    private void initialiseMemberId() {
+        synchronized (mutex) {
```

Comment:
There's `sensors().modify(...)` which will do the same, but I would probably have implemented it similarly to avoid the Java syntax overhead of anonymous classes.